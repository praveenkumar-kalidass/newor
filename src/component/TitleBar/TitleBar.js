import React from 'react';
import PropTypes from 'prop-types';

import { Bar, BarContainer } from './TitleBar.style';

const TitleBar = ({ children, color, styleProps }) => (
  <BarContainer styleProps={styleProps}>
    <Bar color={color} />
    {children}
  </BarContainer>
);

TitleBar.propTypes = {
  children: PropTypes.node.isRequired,
  color: PropTypes.string.isRequired,
  styleProps: PropTypes.shape({}),
};

TitleBar.defaultProps = {
  styleProps: {},
};

export default TitleBar;
