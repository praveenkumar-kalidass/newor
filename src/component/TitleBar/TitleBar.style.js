import styled from 'styled-components/native';
import { Flex, Box } from 'native-base';

export const BarContainer = styled(Flex).attrs(({ styleProps }) => ({
  direction: 'row',
  ...styleProps,
}))``;

export const Bar = styled(Box).attrs(({ color }) => ({
  bg: color,
  mr: '2',
  pr: '2',
}))``;
