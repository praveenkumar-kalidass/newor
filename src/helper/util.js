import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
