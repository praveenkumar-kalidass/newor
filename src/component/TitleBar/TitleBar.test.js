import React from 'react';
import { Text } from 'react-native';
import { render } from '../../test/util';

import TitleBar from './TitleBar';

describe('TitleBar', () => {
  it('should match snapshot', () => {
    const container = render(
      <TitleBar color="white">
        <Text>Test</Text>
      </TitleBar>,
    );

    expect(container).toMatchSnapshot();
  });
});
