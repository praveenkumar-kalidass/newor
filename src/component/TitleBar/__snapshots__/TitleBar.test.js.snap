// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {},
        ],
      ]
    }
    styleProps={Object {}}
  >
    <View
      dataSet={Object {}}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#FFFFFF",
            "marginRight": 8,
            "paddingRight": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    />
    <Text>
      Test
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
